src/crypto.rs

```diff
@@ -15,31 +15,22 @@
 // You should have received a copy of the GNU General Public License
 // along with this program. If not, see <https://www.gnu.org/licenses/>.
 
-use std::ops::Deref;
-
-use lazy_static::lazy_static;
+use chacha20poly1305::{
+    aead,
+    aead::{Aead, NewAead},
+};
+use generic_array::GenericArray;
 use secstr::{SecStr, SecUtf8};
 use serde::{Deserialize, Serialize};
-use sodiumoxide::crypto::{pwhash, secretbox};
 use thiserror::Error;
 
 use crate::pinentry::Pinentry;
 
-lazy_static! {
-    static ref SODIUMOXIDE_INITIALISED: bool = sodiumoxide::init().map(|()| true).unwrap_or(false);
-}
+/// Nonce used for secret box.
+type Nonce = GenericArray<u8, <chacha20poly1305::ChaCha20Poly1305 as aead::Aead>::NonceSize>;
 
-/// Lazily trigger sodiumoxide initialisation.
-///
-/// Panics if `sodiumoxide::init()` fails.
-///
-/// **This function must be called from all places within this module which
-/// could be called with an unitialized `sodiumoxide`.**
-fn ensure_initialised() {
-    if !SODIUMOXIDE_INITIALISED.deref() {
-        panic!("Failed to initialise sodiumoxide")
-    }
-}
+/// 192-bit salt.
+type Salt = [u8; 24];
```

Comment:
`const`'ing the size for `Salt` and `Nonce` seems like it would be helpful to use here and in the below code